-Wunused-lambda-capture warning with clang++ 5.0.1

<!--
Hello and thanks for contributing! To help us diagnose your problem quickly, please:

 - Include a minimal demonstration of the bug, including code, logs, and screenshots.
 - Ensure you can reproduce the bug using the latest release.
 - Only post to report a bug or request a feature; direct all other questions to: https://stackoverflow.com/questions/tagged/mapbox
-->

**Platform:**

OS X

**version:**

https://github.com/mapbox/mapbox-gl-native/commit/553efa38e3591ce62863d4d74222710f8e3c2337

### Steps to trigger behavior

 1. Use clang++ 5.0.1 from mason on OS X
 2. Build with `mkdir build && cd build && cmake ../ && make`
 3. Hit error on new warning (that I presume does not occur with earlier clang++ versions)

### Expected behavior

No warnings/no compiler errors

### Actual behavior

Hit this error:

```
Users/dane/projects/vt-shaver-cpp/deps/mapbox-gl-native/platform/node/src/node_thread_pool.cpp:9:84: error: lambda capture 'this' is not used [-Werror,-Wunused-lambda-capture]
    : queue(new util::AsyncQueue<std::weak_ptr<mbgl::Mailbox>>(uv_default_loop(), [this](std::weak_ptr<mbgl::Mailbox> mailbox) {
                                                                                   ^
1 error generated.
make[2]: *** [CMakeFiles/mbgl-node.dir/platform/node/src/node_thread_pool.cpp.o] Error 1
make[1]: *** [CMakeFiles/mbgl-node.dir/all] Error 2
make: *** [all] Error 2
```

Also, in the node bindings (perhaps unrelated to compiler version):

```
/Users/dane/projects/vt-shaver-cpp/deps/mapbox-gl-native/platform/node/src/node_map.cpp:484:13: error: 'Call' is deprecated [-Werror,-Wdeprecated-declarations]
        cb->Call(1, argv);
            ^
/Users/dane/projects/vt-shaver-cpp/node_modules/nan/nan.h:1568:3: note: 'Call' has been explicitly marked deprecated here
  Call(int argc, v8::Local<v8::Value> argv[]) const {
  ^
/Users/dane/projects/vt-shaver-cpp/deps/mapbox-gl-native/platform/node/src/node_map.cpp:500:13: error: 'Call' is deprecated [-Werror,-Wdeprecated-declarations]
        cb->Call(2, argv);
            ^
/Users/dane/projects/vt-shaver-cpp/node_modules/nan/nan.h:1568:3: note: 'Call' has been explicitly marked deprecated here
  Call(int argc, v8::Local<v8::Value> argv[]) const {
  ^
/Users/dane/projects/vt-shaver-cpp/deps/mapbox-gl-native/platform/node/src/node_map.cpp:505:13: error: 'Call' is deprecated [-Werror,-Wdeprecated-declarations]
        cb->Call(1, argv);
            ^
/Users/dane/projects/vt-shaver-cpp/node_modules/nan/nan.h:1568:3: note: 'Call' has been explicitly marked deprecated here
  Call(int argc, v8::Local<v8::Value> argv[]) const {
  ^
```

--- a/platform/node/src/node_thread_pool.cpp
+++ b/platform/node/src/node_thread_pool.cpp
@@ -6,7 +6,7 @@
 namespace node_mbgl {
 
 NodeThreadPool::NodeThreadPool()
-    : queue(new util::AsyncQueue<std::weak_ptr<mbgl::Mailbox>>(uv_default_loop(), [this](std::weak_ptr<mbgl::Mailbox> mailbox) {
+    : queue(new util::AsyncQueue<std::weak_ptr<mbgl::Mailbox>>(uv_default_loop(), [](std::weak_ptr<mbgl::Mailbox> mailbox) {
         Worker* worker = new Worker(mailbox);
         Nan::AsyncQueueWorker(worker);
     })) {

